R/assocTestAggregate.R

```diff
@@ -33,6 +35,9 @@ setMethod("assocTestAggregate",
               # do we need to match on alleles?
               match.alleles <- any(c("ref", "alt") %in% names(mcols(currentRanges(gdsobj))))
 
+              # check ploidy
+              if (SeqVarTools:::.ploidy(gdsobj) == 1) male.diploid <- FALSE
```

Comment:
@smgogarten should there be the opposite check here? i.e. if ploidy == 2, male.diploid <- TRUE ?

Reply:
Not necessarily. On the X chromosome, often ploidy == 2 because of the PAR, even though the non-PAR variants are still coded as haploid in some cases. Have not yet solved the problem of how to return dosages as non-missing for 1000 genomes variants (coded as 0/. and 1/.), but this should work if the VCF is coded as 0/0 and 1/0.

Reply:
Ah. Good point. I think this looks good then. 

Do we need some sort of warning or disclaimer about this 0/. issue? What do the results end up looking like? From looking at the code, I'm guessing M.count is NA, which makes both freq and MAC also be NA. If that's true, at least it's not giving _wrong_ results.

Reply:
In the case of the 1000 genomes X chrom file, it returns NA only for the males, so all (non-PAR) X chrom analyses will end up treating the males as missing. That is wrong, and I think we need to fix it rather than giving a disclaimer. But the fix needs to be in SeqArray and SeqVarTools rather than GENESIS.